### PR TITLE
Minor improvements from wm4's fork

### PR DESCRIPTION
**Pull request by jeeb**
These are probably the least controversial or hard-to-review/test changes from his fork.

The EDL mp4 DASH hack thing should be tested but otherwise seems rather straightforward.

v4l2, stream_libarchive and the EDL thing are something that could be back-ported.

### CODE REVIEW

File: player/lua/stats.lua [hunk 1]
```diff
@@ -430,6 +430,8 @@ local function add_file(s)
         append_property(s, "media-title", {prefix="Title:"})
     end
 
+    append_property(s, "file-format", {prefix="Format/protocol:"})
```
Comment (sfan5) [inline player/lua/stats.lua]: for consistency this should say `"Format/Protocol:"`

Comment (jeeb) [reply player/lua/stats.lua]: Done.

File: demux/demux.c [hunk 2]
```diff
@@ -97,11 +97,15 @@ struct demux_opts {
 
 #define OPT_BASE_STRUCT struct demux_opts
 
+#define MAX_BYTES MPMIN(INT64_MAX, (size_t)-1 / 2)
```
Comment (sfan5) [inline demux/demux.c]: stdint.h defines `SIZE_MAX`, so no need for the -1 thing

Comment (jeeb) [reply demux/demux.c]: `SIZE_MAX - 1` seems to be utilized as the biggest thing to be able to malloc elsewhere, and then bstr.c seems to utilize `size >= SIZE_MAX / 2`, so I guess the thing depends on if how the option max value handling is done (whether > or >=).

If it's > then we still need the -1 methinks.

Comment (jeeb) [reply demux/demux.c]: OK, so the type of BYTE_SIZE is int64_t, and `(tmp_int > opt->max)` is how it's checked if I'm reading its int64->long long road correctly (`parse_int64` to `parse_longlong` in `m_option.c`). So we need the -1 as far as I can tell?

Comment (jeeb) [reply demux/demux.c]: Switched to `SIZE_MAX / 2` since we compare against INT64_MAX anyways.

**Pull request merged**